### PR TITLE
Introduce IEventSource2 to not break usages of IEventSource

### PR DESCRIPTION
**Pull request by jeffkl**
We can still improve on this by having an ILogger2 and update Logger but this will keep the telemetry stuff in without breaking people who implement IEventSource

### CODE REVIEW

File: src/Utilities/MuxLogger.cs [hunk 1]
```diff
@@ -1341,7 +1346,11 @@ private void UnregisterAllEventHandlers()
                 _eventSourceForBuild.TaskFinished -= _taskFinishedEventHandler;
                 _eventSourceForBuild.TaskStarted -= _taskStartedEventHandler;
                 _eventSourceForBuild.WarningRaised -= _buildWarningEventHandler;
-                _eventSourceForBuild.TelemetryLogged -= _telemetryEventHandler;
+
+                if (_eventSourceForBuild is IEventSource2)
+                {
+                    ((IEventSource2)_eventSourceForBuild).TelemetryLogged -= _telemetryEventHandler;
```
Comment (rainersigwald) [inline src/Utilities/MuxLogger.cs]: I think the usual advice is to do `as` and check for `null`, instead of checking for `is` followed by a cast.

Comment (jeffkl) [inline src/Utilities/MuxLogger.cs]: you got it

**Comment by jeffkl** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR  please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR please
@dotnet-bot test Windows_NT Build for CoreCLR please

**Pull request merged**